### PR TITLE
fix: fp when installing/uploading plugin via zip file

### PR DESCRIPTION
**Pull request by EsadCetiner**

### CODE REVIEW

File: plugins/wordpress-rule-exclusions-before.conf [hunk 1]
```diff
@@ -1120,4 +1120,34 @@ SecRule REQUEST_FILENAME "@endsWith /wp-admin/admin-ajax.php" \
...
+    phase:2,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'wordpress-rule-exclusions-plugin/1.0.1',\
+    chain"
+    SecRule ARGS:action "@streq upload-plugin" \
+        "t:none,\
+        chain"
+        SecRule FILES:pluginzip "@endsWith .zip" \
```
Comment (theseion) [inline plugins/wordpress-rule-exclusions-before.conf:1135]: `FILES` is not considered a collection by the parser. @airween is that a mistake? The documentation isn't clear on that, but I think that all `FILES...` variables should be considered collections by the parser.

File: plugins/wordpress-rule-exclusions-before.conf [hunk 2]
```diff
@@ -1120,4 +1120,34 @@ SecRule REQUEST_FILENAME "@endsWith /wp-admin/admin-ajax.php" \
...
+            "t:none,\
+            ctl:ruleRemoveTargetById=944110;REQUEST_BODY,\
+            ctl:ruleRemoveTargetById=944250;REQUEST_BODY"
+
+# Installing a plugin via an upload zip file
```
Comment (theseion) [inline plugins/wordpress-rule-exclusions-before.conf:1140]:
```suggestion
# Installing a plugin via zip file upload
```

File: tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
+---
+meta:
+  author: "Esad Cetiner"
+  description: "Wordpress Rule Exclusions Plugin"
+  enabled: true
+  name: 9507974.yaml
+tests:
+  - test_title: 9507974-1
+    desc: Installing a plugin via an upload zip file
```
Comment (theseion) [inline tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml:9]:
```suggestion
    desc: Installing a plugin via zip file upload
```

File: tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml
Comment (theseion) [inline tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml:1]: The test doesn't seem to match the rules quite. There should be two tests, AFAICT, one should test the "install now" thing.

**Review by theseion — CHANGES_REQUESTED** [review]

File: tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml
Comment (EsadCetiner) [reply tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml:1]: I can only include a test for rule 9507974, 9507973 handles the plugin upload.

The Install now thing is at the bottom, it's a bit hard to see:
>               Install Now
>              ------WebKitFormBoundarylBwlewqbn4fBRA7B--

Comment (theseion) [reply tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml:1]: I don't understand. Wouldn't you test 9507973 simply by submitting upload multipart data with a `.zip` extension and testing that `944110` and `944250` don't trigger?

Comment (EsadCetiner) [reply tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml:1]: libModSecurity3 is logging the payload as symbols. I can test on Apache if you want tomorrow.

Comment (theseion) [reply tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml:1]: I quickly wrote the test that I had expected to see and now I'm confused. Why are you removing those two `944` rules at all? Does the zip archive contain Java code?

Comment (EsadCetiner) [reply tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml:1]: @theseion I don't remember the specific plugin that caused this false positive, but the rules look pretty simple so it's possible the PHP code looked similar enough to java. I'd provide a test if I could.

Comment (theseion) [reply tests/regression/wordpress-rule-exclusions-plugin/9507974.yaml:1]: 🤷

File: plugins/wordpress-rule-exclusions-before.conf (hunk 1 above)
Comment (theseion) [reply plugins/wordpress-rule-exclusions-before.conf:1135]: Ping @airween.

File: plugins/wordpress-rule-exclusions-before.conf [hunk 4]
```diff
@@ -1120,4 +1120,34 @@ SecRule REQUEST_FILENAME "@endsWith /wp-admin/admin-ajax.php" \
...
+# Installing a plugin via zip file upload
+SecRule REQUEST_FILENAME "@endsWith /wp-admin/update.php" \
+    "id:9507974,\
+    phase:2,\
+    pass,\
+    t:none,\
+    nolog,\
+    ver:'wordpress-rule-exclusions-plugin/1.0.1',\
+    chain"
+    SecRule ARGS:install-plugin-submit "@streq Install Now" \
```
Comment (azurit) [inline plugins/wordpress-rule-exclusions-before.conf:1149]: I think we can remove this condition.

Comment (EsadCetiner) [reply plugins/wordpress-rule-exclusions-before.conf:1149]: It seems a bit overkill since the false positive is exactly with ``Install Now``.

Comment (azurit) [reply plugins/wordpress-rule-exclusions-before.conf:1149]: It is also an exclusion for submit button caption which si probably not used at all by the application.

Comment (EsadCetiner) [reply plugins/wordpress-rule-exclusions-before.conf:1149]: true...

File: plugins/wordpress-rule-exclusions-before.conf (hunk 1 above)
Comment (theseion) [reply plugins/wordpress-rule-exclusions-before.conf:1135]: FYI, a fix for seclan_parser has been merged. We just need to wait for a new release.

Comment (airween) [reply plugins/wordpress-rule-exclusions-before.conf:1135]: It's done.

https://github.com/coreruleset/secrules_parsing/releases/tag/v0.2.11

**Review by theseion — APPROVED** [review]

**Pull request merged**